genie-agent/src/main/java/com/netflix/genie/agent/execution/services/JobFolderArchiveService.java

```diff
@@ -0,0 +1,39 @@
+/*
+ *
+ *  Copyright 2018 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+
+package com.netflix.genie.agent.execution.services;
+
+import com.netflix.genie.agent.execution.ExecutionContext;
+import com.netflix.genie.agent.execution.exceptions.ArchiveException;
+
+/**
+ * Service for archiving the job folder.
+ *
+ * @author standon
+ * @since 4.0.0
+ */
+public interface JobFolderArchiveService {
+
+    /**
+     * Archive the job folder.
+     *
+     * @param executionContext Execution context
+     * @throws ArchiveException if archival fails
+     */
+    void archiveJobFolder(ExecutionContext executionContext) throws ArchiveException;
```

Comment:
Services should not be coupled with execution context. You should pass the relevant bit of the context (source folder, destination URL, etc) rather than the whole thing.